### PR TITLE
Comp 301 versioned docs

### PR DESCRIPTION
**Pull request by freetonik**
This switches the documentation builder to multiversion mode. Details:

- A separate folder of docs is built for each of released versioned (i.e. for each git tag). 
- The sidebar on each documentation page contains links to all other versions.
- Each page of a non-latest version of docs includes a notice banner ("You're reading outdated docs...")
- The previous location of docs root (`build/sphinx/html/index.html`) now redirects to the latest version of docs (e.g. `build/sphinx/html/3.3/index.html`).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,6 @@ test_report.xml
 
 # Autogenerated JSON schema
 /docs/*_schema_*.json
+
+# Autogenerated redirection page
+/docs/_static/index.html
```
Comment (smite) [inline .gitignore:24]: Missing LF here

File: docs/conf.py [hunk 2]
```diff
@@ -12,28 +12,29 @@
...
+default_py_root = os.path.join(os.getcwd(), os.path.dirname(__file__))
+py_path = os.path.join(os.getenv("SPHINX_MULTIVERSION_SOURCEDIR", default=default_py_root), "../src")
 sys.path.insert(0, os.path.abspath(py_path))
 
-
 # -- Project information -----------------------------------------------------
 
 project = 'IQM client'
-copyright = '2021, IQM'
+copyright = '2022, IQM'
```
Comment (smite) [inline docs/conf.py:25]:
```suggestion
copyright = '2021-2022, IQM'
```

File: docs/conf.py [hunk 3]
```diff
@@ -12,28 +12,29 @@
...
 author = 'IQM client developers'
 
 # The short X.Y version.
 version = ''
 # The full version, including alpha/beta/rc tags.
-release = ''
```
Comment (smite) [inline docs/conf.py:28]: Why was this removed?

Comment (freetonik) [reply docs/conf.py]: My mistake. This isn't actually used in the updated doc building process, as `release` value is derived from a tag, but this line (and line 37) should remain for compatibility with non-versioned variant of Sphinx, just in case.

File: setup.cfg [hunk 4]
```diff
@@ -54,6 +54,7 @@ dev =
     tox == 3.25.0
 docs =
     sphinx == 4.5.0
+    sphinx-multiversion-contrib == 0.2.6
```
[the remark this answers is not in the document]
Comment (freetonik) [reply setup.cfg:57]: Yes. 
I'm not sure this is the best idea ever, but the original package is unmaintained since 2020, and some changes were needed in order for our use case. 

The good thing is that the end result is just a set of folders of HTML docs generated at different points in git history (== at various tag references). So, if needed, it won't be hard to migrate to another doc building mechanism in the future.

[the remark this answers is not in the document]
Comment (hay-k) [reply setup.cfg:57]: Should it then be under IQMs GitHub and pypi accounts?

[the remark this answers is not in the document]
Comment (freetonik) [reply setup.cfg:57]: Yes, they are now! :)

**Review by hay-k — APPROVED** [review]
